### PR TITLE
Upload source data before generating mlbf for easier debugging

### PR DESCRIPTION
**Pull request by jschanck**
This splits `workflow/2-upload_artifacts_to_storage` into `workflow/1-upload_data_to_storage` and  `workflow/3-upload_mlbf-to_storage` and makes the former run before the filter generation workflow. The idea here is that if filter generation fails we want to be able to pull the source material from GCP for local debugging.

Also a bit of maintenance here to get a base docker image that can run binaries from recent versions of rust. And some changes to make the logs a bit quieter.

### CODE REVIEW

File: containers/scripts/crlite-generate.sh [hunk 1]
```diff
@@ -43,13 +43,20 @@ ls -latS ${ID}/revoked | head
 echo "crlite-fullrun: disk usage"
 du -hc ${ID}
 
-${workflow}/1-generate_mlbf ${ID} \
+if [ "x${DoNotUpload}x" == "xx" ] ; then
+  echo "archiving crls"
+  tar -czf ${ID}/crls.tar.gz -C ${CRLS} .
+  echo "uploading source materials"
+  ${workflow}/1-upload_data_to_storage ${ID} \
```
Comment (mozkeeler) [inline containers/scripts/crlite-generate.sh:50]: Should this be `1-upload_artifacts_to_storage`? Or is there a missing file or something?

File: hooks/go-no-replacement.sh
Comment (mozkeeler) [inline hooks/go-no-replacement.sh:1]: I think this is still referenced in `[.pre-commit-config.yaml](https://github.com/mozilla/crlite/blob/f1ccc84b92866386ed15bbecf548d22f60c02733/.pre-commit-config.yaml#L30)` - should we just remove that too?

File: workflow/1-upload_artifacts_to_storage
Comment (mozkeeler) [inline workflow/1-upload_artifacts_to_storage:1]: This seems to be largely the same as `[3-upload_mlbf_to_storage](https://github.com/mozilla/crlite/pull/296/files#diff-2af64200fdbc3cf8f17b66f51f3515de134fc381d689ecb1185e100ff63ad816)` - was this intended? Is there a way they could share code?

**Review by mozkeeler — CHANGES_REQUESTED** [review]
I think there might be a missing file or something?

File: containers/scripts/crlite-generate.sh (hunk 1 above)
Comment (jschanck) [reply containers/scripts/crlite-generate.sh:50]: ah, yeah, hg <-> git context switch  mishap.

**Review by mozkeeler — APPROVED** [review]
LGTM! Just the one comment.

**Pull request merged**